sklearn/datasets/_twenty_newsgroups.py

```diff
@@ -76,7 +77,8 @@ def _download_20newsgroups(target_dir, cache_path):
     archive_path = _fetch_remote(ARCHIVE, dirname=target_dir)
 
     logger.debug("Decompressing %s", archive_path)
-    tarfile.open(archive_path, "r:gz").extractall(path=target_dir)
+    with tarfile.open(archive_path, "r:gz") as fp:
+        tarfile_extractall(fp, path=target_dir)
```

Comment:
This line being not covered was not covered before either in a PR (unless you triggered a scipy-dev build).

Any issues will be caught in a scheduled CI run on `main`